### PR TITLE
Refactoring of sequence testing code

### PR DESCRIPTION
**Pull request by grafnu**
Should be functionality neutral -- no new tests or capabilities.

### CODE REVIEW

File: validator/src/main/java/com/google/daq/mqtt/validator/SequenceValidator.java [hunk 1]
```diff
@@ -651,25 +658,34 @@ private void handleDeviceMessage(Map<String, Object> message, String subFolderRa
...
+        debug("Updated config:\n" + OBJECT_MAPPER.writeValueAsString(converted));
+        recordDeviceConfig();
+      } else if (converted instanceof State) {
+        debug("Updated state:\n" + OBJECT_MAPPER.writeValueAsString(converted));
+        deviceState = (State) converted;
+        validSerialNo();
+        info("Updated state has last_config " + getTimestamp(
+            ((State) converted).system.last_config));
+      } else {
+        warning("Unknown update type " + converted.getClass().getSimpleName());
```
Comment (johnrandolph) [inline validator/src/main/java/com/google/daq/mqtt/validator/SequenceValidator.java:685]: Is this severe enough?

File: validator/src/main/java/com/google/daq/mqtt/validator/SequenceValidator.java [hunk 2]
```diff
@@ -424,8 +438,9 @@ private void queryState() {
    */
   @After
   public void tearDown() {
-    if (traceLogLevel()) {
-      warning("Not resetting config b/c trace is enabled.");
+    recordMessages = false;
+    if (debugLogLevel()) {
+      warning("Not resetting config because debug is enabled.");
```
Comment (johnrandolph) [inline validator/src/main/java/com/google/daq/mqtt/validator/SequenceValidator.java:443]: It isn't clear at first glance why you don't reset the config in this case. Can you add a comment to quickly spell it out?

File: validator/src/main/java/com/google/daq/mqtt/validator/Validator.java [hunk 3]
```diff
@@ -114,18 +114,18 @@ public class Validator {
...
   public Validator(String projectId) {
     this.projectId = projectId;
   }
 
   /**
-   * Instantiate a stream validator instance.
+   * Let's go.
    *
-   * @param args What to do!
+   * @param args This is the way
```
[the remark this answers is not in the document]
Comment (grafnu) [reply validator/src/main/java/com/google/daq/mqtt/validator/Validator.java:128]: What? I wasn't planning on anybody actually *reading* my comments... that's not fair!  (But updated to something more sensical)

File: validator/src/main/java/com/google/daq/mqtt/validator/SequenceValidator.java (hunk 1 above)
Comment (grafnu) [reply validator/src/main/java/com/google/daq/mqtt/validator/SequenceValidator.java:685]: updated

**Pull request merged**